diff --git a/dist/index.esm.js b/dist/index.esm.js
index 0992869f2bc525d938ad60815cd0630895a02bac..6cc84a9809f282ee0b9e1f2e5563486a567b5d3f 100644
--- a/dist/index.esm.js
+++ b/dist/index.esm.js
@@ -83,8 +83,15 @@ const BLACK_COLUMN_VALUES = {
     g: 1,
     h: 0,
 };
-const WHITE_ROWS = [2, 1, 0];
-const BLACK_ROWS = [0, 1, 2];
+const WHITE_ROWS = import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" ? [2, 1, 0] : 
+    import.meta.env.VITE_USE_BANQUET_INDICES === "true" ? [7, 6, 5] : 
+    [7, 6, 5, 4, 3, 2, 1, 0];
+const BLACK_ROWS = import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" ? [0, 1, 2] : 
+    import.meta.env.VITE_USE_BANQUET_INDICES === "true" ? [5, 6, 7] 
+    : [0, 1, 2, 3, 4, 5, 6, 7];
+
+const BOARD_SIZE = (import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" || import.meta.env.VITE_USE_BANQUET_INDICES  === "true") ? 3 : 8;
+const BOARD_OFFSET = import.meta.env.VITE_USE_BANQUET_INDICES  === "true" ? 5 : import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" ? 1 : 0;
 
 // https://commons.wikimedia.org/wiki/Category:SVG_chess_pieces
 // By en:User:Cburnett - Own work
@@ -4844,15 +4851,15 @@ function Squares() {
         });
         return result;
     }, [premoves]);
-    return (jsx("div", Object.assign({ "data-boardid": id }, { children: [...Array(3)].map((_, r) => {
+    return (jsx("div", Object.assign({ "data-boardid": id }, { children: [...Array(BOARD_SIZE)].map((_, r) => {
             return (jsx("div", Object.assign({ style: {
                     display: "flex",
                     flexWrap: "nowrap",
                     width: boardWidth,
-                } }, { children: [...Array(3)].map((_, c) => {
+                } }, { children: [...Array(BOARD_SIZE)].map((_, c) => {
                     const square = boardOrientation === "black"
-                        ? (COLUMNS[2 - c] + (r + 2))
-                        : (COLUMNS[c] + (4 - r));
+                        ? (COLUMNS[BOARD_SIZE - 1 - c] + (r + 1 + BOARD_OFFSET))
+                        : (COLUMNS[c] + (BOARD_SIZE - r + BOARD_OFFSET));
                     const squareColor = c % 2 === r % 2 ? "white" : "black";
                     const squareHasPremove = premoves.find((p) => p.sourceSq === square || p.targetSq === square);
                     const squareHasPremoveTarget = premovesHistory
diff --git a/dist/index.js b/dist/index.js
index fddcb4cc3fdd5ff2697ab748afc24bf4a58da8fb..06837247f5dc477684b337c2aec9975da99a7420 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -30,6 +30,9 @@ function __rest(s, e) {
     return t;
 }
 
+const BOARD_SIZE = (import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" || import.meta.env.VITE_USE_BANQUET_INDICES  === "true") ? 3 : 8;
+const BOARD_OFFSET = import.meta.env.VITE_USE_BANQUET_INDICES  === "true" ? 5 : 0;
+
 const COLUMNS = "abc".split("");
 const START_POSITION_OBJECT = {
     a8: "bR",
@@ -69,14 +72,29 @@ const WHITE_COLUMN_VALUES = {
     a: 0,
     b: 1,
     c: 2,
+    d: 3,
+    e: 4,
+    f: 0,
+    g: 1,
+    h: 2,
 };
 const BLACK_COLUMN_VALUES = {
-    a: 2,
-    b: 1,
-    c: 0,
+    a: 7,
+    b: 6,
+    c: 5,
+    d: 4,
+    e: 3,
+    f: 2,
+    g: 1,
+    h: 0,
 };
-const WHITE_ROWS = [2, 1, 0];
-const BLACK_ROWS = [0, 1, 2];
+
+const WHITE_ROWS = import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" ? [2, 1, 0] : 
+    import.meta.env.VITE_USE_BANQUET_INDICES === "true" ? [7, 6, 5] : 
+    [7, 6, 5, 4, 3, 2, 1, 0];
+const BLACK_ROWS = import.meta.env.VITE_USE_HEXAPAWN_INDICES  === "true" ? [0, 1, 2] : 
+    import.meta.env.VITE_USE_BANQUET_INDICES === "true" ? [5, 6, 7] 
+    : [0, 1, 2, 3, 4, 5, 6, 7];
 
 // https://commons.wikimedia.org/wiki/Category:SVG_chess_pieces
 // By en:User:Cburnett - Own work
@@ -4836,15 +4854,15 @@ function Squares() {
         });
         return result;
     }, [premoves]);
-    return (jsxRuntime.jsx("div", Object.assign({ "data-boardid": id }, { children: [...Array(3)].map((_, r) => {
+    return (jsxRuntime.jsx("div", Object.assign({ "data-boardid": id }, { children: [...Array(BOARD_SIZE)].map((_, r) => {
             return (jsxRuntime.jsx("div", Object.assign({ style: {
                     display: "flex",
                     flexWrap: "nowrap",
                     width: boardWidth,
-                } }, { children: [...Array(3)].map((_, c) => {
+                } }, { children: [...Array(BOARD_SIZE)].map((_, c) => {
                     const square = boardOrientation === "black"
-                        ? (COLUMNS[2 - c] + (r + 1))
-                        : (COLUMNS[c] + (3 - r));
+                        ? (COLUMNS[BOARD_SIZE - 1 - c] + (r + 1 + BOARD_OFFSET))
+                        : (COLUMNS[c] + (BOARD_SIZE - r + BOARD_OFFSET));
                     const squareColor = c % 2 === r % 2 ? "white" : "black";
                     const squareHasPremove = premoves.find((p) => p.sourceSq === square || p.targetSq === square);
                     const squareHasPremoveTarget = premovesHistory
